File: Python/797 All Paths From Source to Target.py
class Solution(object):
    def allPathsSourceTarget(self, graph):
        """
        :type graph: List[List[int]]
        :rtype: List[List[int]]
        """
        ans = []
        stack = [(graph[0],[])]
        while stack:
            curr, route = stack.pop()
            for ele in curr:
                if ele == len(graph) - 1:
                    ans.append(route + [len(graph) - 1])
                    continue
                stack.append((graph[ele], route + [ele]))
        return ans

File: C++/797 All Paths From Source to Target.cpp
class Solution {
public:
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        vector<vector<int>> ans;
        vector<int> tmp;
        dfs(graph, ans, tmp, 0);
        return ans;
    }
    
    void dfs(vector<vector<int>>& graph, vector<vector<int>>& paths, vector<int> path, int curr) {
        path.push_back(curr);
        if (curr == graph.size() - 1) paths.push_back(path);
        else for (int nei : graph[curr]) dfs(graph, paths, path, nei);
    }
};